src/Microsoft.AspNetCore.WebHooks.Receivers/Extensions/WebHookReceiverConfigurationRootExtensions.cs

```diff
@@ -0,0 +1,107 @@
+
+using System;
+using System.ComponentModel;
+using Microsoft.AspNetCore.WebHooks;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Extension methods for <see cref="IConfigurationRoot"/>.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class WebHookReceiverConfigurationRootExtensions
+    {
+        /// <summary>
+        /// Gets the secret key <see cref="IConfigurationSection"/> with the given <paramref name="sectionKey"/> and
+        /// <paramref name="id"/>.
+        /// </summary>
+        /// <param name="configuration">The <see cref="IConfigurationRoot"/>.</param>
+        /// <param name="sectionKey">
+        /// The key (relative to <see cref="WebHookConstants.ReceiverConfigurationSectionKey"/>) of the
+        /// <see cref="IConfigurationSection"/> containing the receiver-specific
+        /// <see cref="WebHookConstants.SecretKeyConfigurationKeySectionKey"/> <see cref="IConfigurationSection"/>.
+        /// Typically this is the name of the receiver e.g. <c>github</c>.
+        /// </param>
+        /// <param name="id">
+        /// The (potentially empty) key (relative to <see cref="WebHookConstants.ReceiverConfigurationSectionKey"/>,
+        /// <paramref name="sectionKey"/> and <see cref="WebHookConstants.SecretKeyConfigurationKeySectionKey"/>) of
+        /// the <see cref="IConfigurationSection"/> to return. This allows an <see cref="IWebHookReceiver"/> to support
+        /// multiple WebHook senders with individual
+        /// configurations.
+        /// </param>
+        /// <returns>
+        /// The secret key <see cref="IConfigurationSection"/> with the given <paramref name="sectionKey"/> and
+        /// <paramref name="id"/>. <see langword="null"/> if the <see cref="IConfigurationSection"/> does not exist.
+        /// </returns>
+        public static IConfigurationSection GetSecretKeys(
+            this IConfigurationRoot configuration,
+            string sectionKey,
+            string id)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+            if (sectionKey == null)
+            {
+                throw new ArgumentNullException(nameof(sectionKey));
+            }
+
+            if (string.IsNullOrEmpty(id))
+            {
+                id = WebHookConstants.DefaultIdConfigurationKey;
+            }
+
+            // Look up configuration value for these keys.
+            var key = ConfigurationPath.Combine(
+                WebHookConstants.ReceiverConfigurationSectionKey,
+                sectionKey,
+                WebHookConstants.SecretKeyConfigurationKeySectionKey,
+                id);
+
+            var section = configuration.GetSection(key);
+            if (!section.Exists())
+            {
+                return null;
+            }
```

Comment:
It really seems like we should follow the conventions of IConfiguration here. If IConfiguration uses the null-object-pattern then we should as well.